### PR TITLE
Cleanup src

### PR DESCRIPTION
**Pull request by thomaspinder**
## Type of changes

- [X] Bug fix
- [ ] New feature
- [ ] Documentation / docstrings
- [ ] Tests
- [ ] Other

## Checklist

- [x] I've formatted the new code by running `poetry run pre-commit run --all-files --show-diff-on-failure` before committing.
- [ ] I've added tests for new code.
- [ ] I've added docstrings for the new code.

## Description

Tighten up the underlying codebase by removing aspects of the code that are either incongruent with the entire API, or introduce too much boilerplate code for their returned utility. The code being removed is deeply woven into the source code, so adding deprecations is not possible.

Notes:
- Once approved, I'll open an issue to refactor the pathwise sampling notebook with masking removed.
- The `strict=True` argument has been added to all `zip` calls to make Ruff compliant with Python 3.10 standards. See [B905](https://docs.astral.sh/ruff/rules/zip-without-explicit-strict/)

Issue Number: N/A

### CODE REVIEW

File: docs/examples/graph_kernels.py [hunk 1]
```diff
@@ -158,6 +174,7 @@
     model=posterior,
     objective=jit(gpx.ConjugateMLL(negative=True)),
     train_data=D,
+    max_iters=1000
```
Comment (henrymoss) [inline docs/examples/graph_kernels.py:177]: did this fix it? Seems like a huge number for LBFGS

File: gpjax/gps.py [hunk 2]
```diff
@@ -495,66 +478,41 @@ def predict(
...
+        covariance = Ktt - jnp.matmul(Kxt.T, Sigma_inv_Kxt)
         covariance += cola.ops.I_like(covariance) * self.prior.jitter
         covariance = cola.PSD(covariance)
-        rval = GaussianDistribution(jnp.atleast_1d(mean.squeeze()), covariance)
-        if m == 1:
-            return rval
-        else:
-            return ReshapedDistribution(rval, (n_test, m))
+
+        return GaussianDistribution(jnp.atleast_1d(mean.squeeze()), covariance)
```
Comment (henrymoss) [inline gpjax/gps.py:515]: Readable!

**Review by henrymoss — APPROVED** [review]
Nice bit of surgery Dr. Pinder

File: docs/examples/graph_kernels.py (hunk 1 above)
Comment (thomaspinder) [reply docs/examples/graph_kernels.py:177]: No - the error persists. Works locally
